### ISSUE TITLE
Default clone should use "--depth 1"

### ISSUE THREAD

**Issue by dietmarw** [drone/drone#1229]
For normal CI purposes it does not make sense to clone the whole history of a repo. So the default clone should be using the `--depth 1` flag in order to only retrieve the latest revision. Or at least add an option to do so.

**Comment by RomanSaveljev** [comment]
:+1:

**Comment by RomanSaveljev** [comment]
@dietmarw I was just reading documents from the upcoming 0.4.0 and you could configure the cloning depth in your `.drone.yml` there.

Please see https://github.com/drone/drone/blob/0.4.0/docs/build/clone.md

**Comment by dietmarw** [comment]
@RomanSaveljev Thanks for the tip. But I was confusing this issue tracker with the drone.io service. So it's nice to know that drone the software will support this but will be curious when drone.io the service will be offering the same feature.

**Comment by RomanSaveljev** [comment]
I have tried playing with the drone.io before switching to a self-hosted installation. I think this is the most accessible way to use the drone for the time being.

**Comment by bradrydzewski** [comment]
@dietmarw the good news is this open source codebase will replace and power the drone.io service when it becomes more stable. Unfortunately I don't have good timing estimates to provide. It largely depends on the success of this upcoming release and if the experimental features we've introduced (plugins, matrix, etc) sufficiently meet the expectations of the community or require us to go back to the drawing board and re-design.

I expect we'll know a bit more in a few weeks, once people have gotten a chance to use 0.4 and we've collected sufficient feedback to put together a roadmap. Definitely check back on our Gitter channel in a few weeks and we can give an update.

Closing this issue since 0.4 does support `--depth`
